lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -1158,6 +1164,67 @@ void amf_app_handle_resource_setup_response(
   }
 }
 
+/* Handling PDU Session Resource Modify Response sent from gNB*/
+void amf_app_handle_resource_modify_response(
+    itti_ngap_pdu_session_resource_modify_response_t session_mod_resp) {
+  amf_ue_ngap_id_t ue_id;
+
+  ue_m5gmm_context_s* ue_context = nullptr;
+  std::shared_ptr<smf_context_t> smf_ctx;
+
+  /* Check if failure message is not NULL and if NULL,
+   * it is successful message from gNB.
+   * Nothing to in this case. If failure message comes from gNB
+   * AMF need to report this failed message to SMF
+   *
+   * NOTE: only handling success part not failure part
+   * will be handled later
+   */
+  OAILOG_DEBUG(LOG_AMF_APP,
+               "Handling uplink PDU session modify response message\n");
+
+  if (session_mod_resp.pduSessResourceModRespList.no_of_items > 0) {
+    ue_id = session_mod_resp.amf_ue_ngap_id;
+
+    ue_context = amf_ue_context_exists_amf_ue_ngap_id(ue_id);
+    if (ue_context == NULL) {
+      OAILOG_ERROR(LOG_AMF_APP,
+                   "UE context not found for the ue_id = " AMF_UE_NGAP_ID_FMT,
+                   ue_id);
+      return;
+    }
+
+    smf_ctx = amf_get_smf_context_by_pdu_session_id(
+        ue_context,
+        session_mod_resp.pduSessResourceModRespList.item[0].Pdu_Session_ID);
+    if (smf_ctx == NULL) {
+      OAILOG_ERROR(
+          LOG_AMF_APP, "PDU session  not found for session_id = %lu\n",
+          session_mod_resp.pduSessResourceModRespList.item[0].Pdu_Session_ID);
+      return;
+    }
+
+    /* This is success case and we need not to send message to SMF
+     * and drop the message here
+     */
+    amf_ue_ngap_id_t ue_id;
+    ue_m5gmm_context_s* ue_context = nullptr;
+    ue_id = session_mod_resp.amf_ue_ngap_id;
+
+    ue_context = amf_ue_context_exists_amf_ue_ngap_id(ue_id);
+    // Handling of ue context
+    if (!ue_context) {
+      OAILOG_ERROR(LOG_AMF_APP,
+                   "UE context not found for UE ID: " AMF_UE_NGAP_ID_FMT,
+                   ue_id);
+    }
+  } else {
+    // TODO: implement failure message from gNB. messagge to send to SMF
```

Comment:
What is the expected response in case of gNB failure for PDU session modify response. This is TODO currently.

Reply:
send negative response to sessiond, but currently negative scenarios not planned